nilearn/experimental/plotting/_surface_plotting.py

```diff
@@ -0,0 +1,268 @@
+"""Experimental Surface plotting."""
+
+from __future__ import annotations
+
+import numpy
+
+from nilearn import plotting as old_plotting
+from nilearn._utils import fill_doc
+from nilearn.experimental.surface import Mesh, PolyMesh, SurfaceImage
+
+
+@fill_doc
+def plot_surf(
+    surf_map: SurfaceImage | str | numpy.array | None = None,
+    hemi: str = "left",
+    surf_mesh: (
+        str | list[numpy.array, numpy.array] | Mesh | PolyMesh | None
+    ) = None,
```

Comment:
given that currently surface_mesh can be a str pointing to a file containing a mesh, I think it may be appropriate to also be able to pass a SurfaceImage to surf_mesh

like is done for bg_map

Reply:
Note sure I follow you here.

Reply:
Not 100% sure about this, so to be discussed but it seems that it would convenient if users could directly pass a SurfaceImage instance to this parameter and let nilearn get the Polymesh from the instance rather than forcing users to pass just a Mesh / PolyMesh.

I may be overthinking this but if users can just deal with fewer types of objects (SurfaceImage).

Reply:
it would be weird to pass a different image than the one we are plotting just to grab the mesh from it. either the mesh is `None` in which case we use the mesh that comes with `surf_map`, or we want to provide a different mesh (eg an inflated surface) in which case we pass `surf_mesh`, which should be a mesh. if it happens to be stored as part of another image (which is not very likely if it is an inflated or flat surface used just for plotting) the user can always do `img.mesh` to access it

Reply:
okidoki
not a hill I wanna die on and if users request it we can think about changing that later

but then do you think that it makes sense to be able to pass a SurfaceImage for the background map the way I have done?